### DISCUSSION TITLE
SAY assembler instruction with '\-' to leave out the line end (crlf)

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#283] [Progress Updates]
NetRexx has the convention that on a SAY, when the last character of the line is a NUL ('\-' or '\0'), the linefeed is suppressed. I miss this on other Rexxen where a lot of concatenation needs to take place before we output a line.
Can we consider to add this feature to CREXX?

**Comment by adesutherland** [comment]
As discussed I would rather a new instruction, options:

- OUT
- SAYC
- COUT
- STDOUT
- PARTSAY
- CSAY
- ...

A

**Comment by Peter-Jacob** [comment]
I'll implement a new instruction called SAYX for the moment. I am clueless about names ...

**Comment by Peter-Jacob** [comment]
I added the new instruction sayx which writes a line (string/number) without a linefeed: commit 0735b5a7a9231ba51889244282e272e749c9f73c.  There is currently no REXX statement that compiles into a sayx, you must therefore use the assembler prefix to use it, additionally, a convert to string instruction is necessary (ITOS, etc.)
Example:
```
do k=1 to 5
   i=4711*k
   j=2*k+i
   assembler itos j
   assembler sayx j
end 
```
Result: 47139426141391885223565

To finish the line a simple say ' ' will do.
